kafka-postgres-reader/src/main/java/io/arenadata/kafka/postgres/reader/service/impl/PublishServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2021 Arenadata Software LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.arenadata.kafka.postgres.reader.service.impl;
+
+import io.arenadata.kafka.postgres.avro.codec.AvroEncoder;
+import io.arenadata.kafka.postgres.avro.model.DtmQueryResponseMetadata;
+import io.arenadata.kafka.postgres.reader.model.KafkaBrokerInfo;
+import io.arenadata.kafka.postgres.reader.service.KafkaProducerProvider;
+import io.arenadata.kafka.postgres.reader.service.PublishService;
+import io.vertx.core.Future;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class PublishServiceImpl implements PublishService {
```

Comment:
Please, don't 😃
I bet if you drop this `PublishService` and get rid of this `Impl` for each and every single interface, nothing happens. No errors nor exceptions. Spring won't fail and life will become simpler.

Reply:
Please try to delete the interface and call this service class `PublishService`, i.e. delete this ugly `Impl`. Good time to check if spring proxies could deal with that.